### PR TITLE
Store balance value at BInt.

### PR DESCRIPTION
**Pull request by noppefoxwolf**
Currently, Balance.swift cannot store UInt64 or more.
So, How about use big number integer like BInt?

### CODE REVIEW

**Comment by yuzushioh** [comment]
@noppefoxwolf Yes! I was going to use BigInt instead of UInt64 eventually so thank you for doing it🎉 will review this PR!

File: EthereumKit/Helper/SMP+Extensions.swift [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+//  Created by Tomoya Hirano on 2018/02/22.
+//  Copyright © 2018年 yuzushioh. All rights reserved.
+//
+
+import UIKit
+
+// https://github.com/mkrd/Swift-Big-Integer/pull/24
+extension BInt {
+  public static func make(_ str: String, radix: Int) -> BInt? {
+    let bint16 = BInt(16)
```
[the remark this answers is not in the document]
Comment (yuzushioh) [reply EthereumKit/Helper/SMP+Extensions.swift]: and not sure if this initializer/method should be public-scoped

File: EthereumKit/Helper/SMP.swift [hunk 2]
```diff
@@ -0,0 +1,2514 @@
+//
```
[the remark this answers is not in the document]
Comment (yuzushioh) [reply EthereumKit/Helper/SMP.swift]: It's ok to include Carthage build in the git repository

File: EthereumKit/Helper/SMP.swift [hunk 3]
```diff
@@ -0,0 +1,2514 @@
...
+	//
+	//
+
+	public var description: String
+	{
+		return (self.sign ? "-" : "").appending(self.limbs.decimalRepresentation)
+	}
+
+	public init(number: String, withBase base: Int)
+	{
```
Comment (yuzushioh) [inline EthereumKit/Helper/SMP.swift]: IMO: `number` parameter should be string/numberString😅 its a bit confusing. (no need to fix it)

File: EthereumKit/Networking/Request/RPC.swift [hunk 4]
```diff
@@ -24,10 +24,14 @@ public final class RPC {
...
                 throw JSONRPCError.unexpectedTypeObject(resultObject)
             }
             return Balance(wei: wei)
         }
     }
 }
+
+
+
+
```
Comment (yuzushioh) [inline EthereumKit/Networking/Request/RPC.swift]: can you unstage those empty lines?

**Comment by yuzushioh** [comment]
about BigInt, I was going to use [BigInt](https://github.com/attaswift/BigInt) and is already in the repository actually. I am fine with the one you choose though 👍

File: EthereumKit/Helper/SMP+Extensions.swift [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+}
+
+extension String {
+  func toHex() -> String {
+    var hex = self
+    while hex.first == "0" || hex.first == "x" {
+      hex = String(hex.dropFirst())
+    }
+    return hex
+  }
```
[the remark this answers is not in the document]
Comment (yuzushioh) [reply EthereumKit/Helper/SMP+Extensions.swift]: but whichever is fine with me

**Comment by noppefoxwolf** [comment]
Please check these PRs.
https://github.com/yuzushioh/Swift-Big-Integer/pull/2
https://github.com/yuzushioh/Swift-Big-Integer/pull/3

I will modify Cartfile when able to be add SMP framework with Carthage.

**Comment by yuzushioh** [comment]
@noppefoxwolf 👍 I merged those two PRs.

**Comment by noppefoxwolf** [comment]
@yuzushioh fixed. 🙇

**Comment by yuzushioh** [comment]
↑ I did it 💪🏻

**Pull request merged**